src/agent/solana/exporter.rs

```diff
@@ -501,6 +494,31 @@ impl Exporter {
                     false
                 }
             })
+            .filter(|(id, info)| {
+                // Filtering out prices that are being updated too frequently according to publisher_permission.publish_interval
+                let last_info = self.last_published_state.get(id);
+                if last_info.is_none() {
+                    // No prior data found, letting the price through
+                    return true;
+                }
+                let last_info = last_info.unwrap();
+
+                let key_from_id = Pubkey::from((*id).clone().to_bytes());
+                let publisher_permisssion = self.our_prices.get(&key_from_id);
+                if publisher_permisssion.is_none() {
+                    // Should never happen since we have filtered out the price above
+                    return false;
+                }
+                let publisher_permission = publisher_permisssion.unwrap();
+
+                if let Some(publish_interval) = publisher_permission.publish_interval {
+                    if info.timestamp.saturating_sub(last_info.timestamp) < publish_interval.as_secs() as i64 {
```

Comment:
you are falling back to seconds here too. I recommend changing the info.timestamp and now to `Instant` ([link](https://docs.rs/tokio/latest/tokio/time/struct.Instant.html)) so you get duration_since and can compare properly.

Reply:
So we have a problem here. we are only storing seconds for `last_info.timestamp`. To fix this to work with values <1s we have to change the info logic to store NaiveTime instead of u64. I'll look into that to see how much effort its gonna take.

Reply:
Why not the Instant that I mentioned?

Reply:
Can you explain more how you want to measure the exact time with that? Because we have an exact duration we want to enforce. The Instant create mentions that

> Note, however, that instants are not guaranteed to be steady. In other words, each tick of the underlying clock may not be the same length (e.g. some seconds may be longer than others). An instant may jump forwards or experience time dilation (slow down or speed up), but it will never go backwards.

It seems like it's not for the purpose of calculating exact times. Am I missing something?

Reply:
I tried using `tokio::time::Instant` and `std::time::Instant`. It could have potentially worked since we are only comparing some `Instant::now()` times. But they don't allow access to create or read the underlying time data. In `metrics.rs` we are storing the exact time for the prices and there is no way to convert Instant to a unix timestamp.
I eventually went with NaiveDate .